### ISSUE TITLE
How does the ReturnUri work with android?

### ISSUE THREAD

**Issue by dazinator** [Azure-Samples/active-directory-dotnet-native-multitarget#6]
Firstly - thanks for cross platform! :)

I had a go with getting this working with my android app, but the ReturnUri presents a problem.

So, when the user first loggs in, they are sent off to enter their credentials and that works fine. However because I do not know what ReturnUri to use on android, i'm just using: "http://www.someurl.com" - and becuase of this, as soon as the user logs in, they then get redirected of to this URL - which breaks the flow (because it's obviously not valid) - as a big error comes up on the sign in page.

I know on windows store apps you can use `WebAuthenticationBroker` to get the return url for the current app. What's the way to do this on Android? is there one? Is there a way to not pass a returnurl for android? (but then I am not sure how the saml token is obtained after the user submits their credentials)

Hopefully I am making sense, and hopefully i'm just misunderstanding something simple?

Cheers

**Comment by dazinator** [comment]
I managed to get this to work in the end (somewhat) but I needed to put an IntentFilter attribute above the android Activity class, for the ReturnUri stuff to correctly return to my app after login.

For example, if the return url was: `appschema://myapp` then you'd place this attribute above the activity:

```
 [IntentFilter(new[] { Intent.ActionView }, Categories = new[] { Intent.CategoryDefault, Intent.CategoryBrowsable }, DataScheme = "appschema", DataHost = "myapp")]
    [Activity(Label = "myapp", MainLauncher = true, Icon = "@drawable/icon")]
    public class LoginActivity : Activity
    {

```
You may want to update the sample with that.,.

**Comment by dstrockis** [comment]
Thanks @dazinator... Sorry for your troubles.  This has worked for me in the past without the intent filter, but I'll investigate and see if I can repro.

**Comment by btitular** [comment]
I'm using the most recent library (3.3 alpha).

I was also having a problem with app returning from the authentication phase. It kept giving me an error that it could not resolve the web page. I then got a clue from: 

https://stackoverflow.com/questions/30048166/adal-3-2-alpha-acquiretokenasync-ios-does-not-dismiss-sign-in-form?newreg=ef8573f754d54ff2a2bc0c6f9c4274c0

Afshin mentioned a possible case sensitive issue. Sure enough, as soon as I changed my redirect url to all lower case in the PCL, the program works as expected.
